crates/engine/src/rocksdb_engine/mod.rs

```diff
@@ -173,49 +198,53 @@ impl StorageEngine for RocksEngine {
     }
 
     #[inline]
-    fn write_batch(&self, wr_ops: Vec<WriteOperation<'_>>, sync: bool) -> Result<(), EngineError> {
-        let mut batch = WriteBatchWithTransaction::<false>::default();
+    fn write_batch(&self, wr_ops: Vec<WriteOperation<'_>>, _sync: bool) -> Result<(), EngineError> {
+        let transaction = self.inner.transaction();
         let mut size = 0;
+        #[allow(clippy::pattern_type_mismatch)] // can't be fixed
```

Comment:
It can be fixed.
```rust
 let kvs: Vec<_> = transaction
 .iterator_cf(&cf, mode)
 .take_while(|res| res.as_ref().is_ok_and(|kv| kv.0.as_ref() < to))
 .collect::<Result<Vec<_>, _>>()?;
```

Reply:
fixed